server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueue.java

```diff
@@ -71,28 +75,36 @@
         RabbitMQMailQueue create(MailQueueName mailQueueName) {
             mailQueueView.initialize(mailQueueName);
 
-            return new RabbitMQMailQueue(metricFactory, mailQueueName,
+            return new RabbitMQMailQueue(
+                metricFactory,
+                mailQueueName,
+                gaugeRegistry,
                 new Enqueuer(mailQueueName, rabbitClient, mimeMessageStore, mailReferenceSerializer,
                     metricFactory, mailQueueView, clock),
                 new Dequeuer(mailQueueName, rabbitClient, mailLoader, mailReferenceSerializer,
-                    metricFactory, mailQueueView), mailQueueView);
+                    metricFactory, mailQueueView),
+                mailQueueView);
         }
     }
 
     private final MailQueueName name;
     private final MetricFactory metricFactory;
+    private final GaugeRegistry gaugeRegistry;
     private final Enqueuer enqueuer;
     private final Dequeuer dequeuer;
     private final MailQueueView mailQueueView;
 
     RabbitMQMailQueue(MetricFactory metricFactory, MailQueueName name,
-                      Enqueuer enqueuer, Dequeuer dequeuer,
+                      GaugeRegistry gaugeRegistry, Enqueuer enqueuer, Dequeuer dequeuer,
                       MailQueueView mailQueueView) {
         this.metricFactory = metricFactory;
         this.name = name;
+        this.gaugeRegistry = gaugeRegistry;
         this.enqueuer = enqueuer;
         this.dequeuer = dequeuer;
         this.mailQueueView = mailQueueView;
+
+        this.gaugeRegistry.register(QUEUE_SIZE_METRIC_NAME_PREFIX + name, this::getSize);
```

Comment:
you should not use string concatenation on anything but string. Here, you actually call MailQueueName.toString